scripts/lib/schemata.py

```diff
@@ -0,0 +1,111 @@
+from operator import itemgetter
+
+from schema import And, Or, Regex, Schema
+
+
+def is_sorted_versions(versions):
+    sorted_versions = sorted(versions, key=itemgetter("version"))
+    return versions == sorted_versions
+
+
+def is_value_unique(key):
+    def fn(data):
+        values = [v[key] for v in data]
+        return len(values) == len(set(values))
+    fn.__name__ = "is_{value}_unique".format(value=key)
+    return fn
+
+
+def is_version_ids_unique(releases):
+    is_version_unique = is_value_unique("version")
+    return is_version_unique((v for r in releases for v in r["versions"]))
+
+
+def is_series_names_unique(releases):
+    series = [r["series"] for r in releases]
+    return len(series) == len(set(series))
+
+
+sha_regex = r"^[0-9a-f]{40}$"
+version_regex = (
+    r"(?P<version>r?[0-9]+\.[0-9]+\.[0-9]+(-(alpha|beta)\.[0-9]+)?)"
+)
+
+version_id_schema = Regex(r"^{vr}$".format(vr=version_regex))
+version_sha_schema = Regex(sha_regex)
+version_schema = Schema(
+    {
+        "version": version_id_schema,
+        "sha": version_sha_schema,
+    },
+)
+
+repo_url_schema = Regex(r"^https://github.com/rcbops/.+")
```

Comment:
We may want to relax this schema to `Regex(r"^https://github.com/.+/.+")` until this code merges into rcbops.